### ISSUE TITLE
1.6 breaks code thunk return type when rejectWithValue is used

### ISSUE THREAD

**Issue by bfricka** [reduxjs/redux-toolkit#1456]
Prior to 1.6, returning `rejectWithValue` (evaluates to `RejectValue<unknown>`, if it's not explicitly typed), would not change the return type of the `AsyncThunk`. For example:

```ts
const fetchFoo = createAsyncThunk('foo', async (_: void, { dispatch, rejectWithValue }) => {
  try {
    return await dispatch(fetch<Foo>())
  } catch (e) {
    return rejectWithValue(e)
  }
})

// Then
try {
  const foo = await dispatch(fetchFoo()).then(unwrapResult) // 1.5.x: Foo, 1.6.x: unknown
} catch (e) {
  // Do something with the actual rejected value
   console.log(e.body?.code)
}
```
This issue persists even after applying the https://github.com/reduxjs/redux-toolkit/pull/1449 fix.

I'm not sure exactly what's going on here, because I honestly don't understand the magic behind the `AsyncThunkPayloadCreatorReturnValue`, neither in 1.5 nor 1.6. It looks like `unwrapResult` and its associated types didn't change at all.

**Comment by phryneas** [comment]
Could you open a PR with a failing type test for this? That would make it easier to fix it.

**Comment by markerikson** [comment]
Available in the 1.7 betas.
